### ISSUE TITLE
release: 21.2 microbenchmark regression investigation and sign-off

### ISSUE THREAD

**Issue by cucaroach** [cockroachdb/cockroach#71820]
This is a tracking bug for issues we want to investigate related to performance for 21.2 similar to what's been done for prior releases (https://github.com/cockroachdb/cockroach/issues/62212).

### SQL

pkg/sql/...  
[spread sheets](https://docs.google.com/spreadsheets/d/1ii8k2fMsPp4sx7i2npdjgU-j3LxfwmoOvEHvreG-Q28/edit?usp=sharing)

pkg/bench/... 
pkg/col/...  
[spread sheets](https://docs.google.com/spreadsheets/d/1QFRgQUqVn1gM_fsC7ex69GUC4FdZHX6qqzzkzQKBgWg/edit#gid=5)

### KV

./pkg/gossip/...
./pkg/roachpb
./pkg/rpc/...
./pkg/server/...

[spread sheets](https://docs.google.com/spreadsheets/d/1RVhSgHJzJg_IXB2sWJbJnoCJawsZwGDN35QC-uk42R4/edit)

./pkg/kv/...
[spread sheets](https://docs.google.com/spreadsheets/d/1NSv_dgchFN-dBTxFcHDHojBA8JrJNXxvkmdNy9xyul4/edit?usp=sharing)

### Bulk I/O

pkg/blobs
[spread sheets](https://docs.google.com/spreadsheets/d/1RZ85VR2Gst5HH0h0H9CLW9hrwn_lo1jvnjH8Ytsik-g/edit)

### Storage

pkg/ccl/storageccl/...
[spread sheets](https://docs.google.com/spreadsheets/d/1uD9XHMT_JtcZfnjiVbspdviI-NVU5I0JA3oW9SAJmQA/edit)

### Misc

./pkg/base ./pkg/ccl/utilccl/... ./pkg/ccl/workloadccl/... ./pkg/cli/... ./pkg/clusterversion ./pkg/cmd/... ./pkg/internal/... ./pkg/security ./pkg/settings ./pkg/testutils/... ./pkg/ts/... ./pkg/util/... ./pkg/workload/...

[spread sheets](https://docs.google.com/spreadsheets/d/1lP4RBjkQVkbv-di4Cwhxvi5q3Rdiq34G5ZFblROUwJQ/edit#gid=1)

### Issues:

- [x] #72110
- [ ] ./pkg/storage/... hangs
- [ ] Why isn't ./pkg/blobs working?
- [ ] Why is ./pkg/bench/... missing a bunch of benchmarks?
- [x] https://github.com/cockroachdb/cockroach/issues/71881
- [x] https://github.com/cockroachdb/cockroach/issues/71886
- [x] https://github.com/cockroachdb/cockroach/issues/72001
- [x] BenchmarkBTreeDeleteInsertCloneEachTime looks wonky (this is why: https://github.com/cockroachdb/cockroach/pull/62294)

Jira issue: CRDB-10784

**Comment by yuzefovich** [comment]
Yeah, there are definitely some regressions in `./pkg/bench`: [here](https://gist.github.com/yuzefovich/23b9354d06750b7157d24203170108b2) is the output of running `PKG=./pkg/bench/... BENCHES=BenchmarkS/MultinodeCockroach` between release-21.1 and release-21.2. And one regression is identified [here](https://github.com/cockroachdb/cockroach/pull/67720#issuecomment-960327536).

I think that it might be worthwhile to wait for all pending fixes to merge to 21.2 branch and then run the benchdiff on `./pkg/bench`.

**Comment by cucaroach** [comment]
@yuzefovich thanks for doing that!   I don't know why my ./pkg/bench run didn't include all those benchmarks.   I did have some problems with some packages where benchdiff was apparently eating the results (ie pkg/blobs).   I need to circle back and see what's going on.   I'll add looking into ./pkg/bench to the list.

**Comment by yuzefovich** [comment]
I finally was able to use `benchdiff` for `./pkg/bench/...`, and the results are of 858eb555aa (21.1) vs f26513f0b0 (21.2) [here](https://gist.github.com/yuzefovich/27d9b8c95ced81d51ad61ebb5765b151). (I only added a single commit to remove `BenchmarkTracing` because it's broken on 21.1 branch. Also, `benchdiff` would always get stuck after some iterations, so I never got 10 successful iterations, and I ended up getting a comparison with `--count=5`.)

The most egregious regression is on some `ScanFilter` in some cases (identified [here](https://github.com/cockroachdb/cockroach/pull/67720#issuecomment-960327536)), but it might be still helpful to spend some time on a Friday to investigate others.
